Create overview page for the CLI

coverbeck opened the pull request:
Gives a high-level overview of the CLI.

Given the potentially uncertain future of the CLI, I didn't want to delve into all the details of what it can do, and showed how the user can discover its features.

dockstore/dockstore#4835
DOCK-2127

denis-yuen reviewed (approved):
ok in theory but build needs work

denis-yuen commented:
> For example, if the PR reviewers decid this new page shouldn't exist, then we don't want to create the topic.

I don't remember what we decided, ok with whatever we wrote down. Or if we didn't maybe it should be in the PR template.

kathy-t commented on docs/advanced-topics/dockstore-cli/overview.rst:
```diff
@@ -0,0 +1,135 @@
+    --ignore-checksums                  Allows you to ignore validating checksums of each downloaded descriptor
+
+
+Common Executed Commands
```
Common Executed Commands -> Commonly Executed Commands?

coverbeck commented:
Updated PR template; removed Hubflow links, see Slack [discussion](https://ucsc-gi.slack.com/archives/C05EZH3RVNY/p1704926441330179). Alternatively, we could link to our copy of Hubflow, but I don't think we want to be responsible for that.

denis-yuen commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -0,0 +1,10 @@
+Please make sure that you've checked the following before submitting your pull request. Thanks!
+
+- [ ] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
+- [ ] If the PR has any new pages, only **AFTER** the PR is approved by all reviewers, generate a new discourse topic, by running the action at https://github.com/dockstore/dockstore-documentation/actions/workflows/add-discourse-topic.yml. This is irreversible, which is why you want to ensure the new pages are approved before generating the topic.
```
Is it irreversible?
I'd kind of assumed that all community leads could delete topics on discourse anyway, in which case creating it before would expose any other kind of build issues with new pages

denis-yuen replied:
That said, clicking into the build I can see the link check and build worked
If we stick with this approach, suggesting new ticket for @ll5zh to split out just the discourse topic check into a new build so we can see at a glance the rest of the build succeeded

coverbeck replied:
> Is it irreversible? I'd kind of assumed that all community leads could delete topics on discourse anyway, in which case creating it before would expose any other kind of build issues with new pages

Irreversible was too strong a word. I'll update the text.

coverbeck replied:
> That said, clicking into the build I can see the link check and build worked If we stick with this approach, suggesting new ticket for @ll5zh to split out just the discourse topic check into a new build so we can see at a glance the rest of the build succeeded

SEAB-6142
